layout/_third-party/math/mathjax.swig

```diff
@@ -1,53 +1,44 @@
-<script type="text/x-mathjax-config">
-  {%- if theme.math.mathjax.mhchem %}
-    MathJax.Ajax.config.path['mhchem'] = '{{ mhchem_uri }}';
-  {%- endif %}
+{%- set mathjax_uri = theme.vendors.mathjax or '//cdn.jsdelivr.net/npm/mathjax@3.0.1/es5/tex-mml-chtml.js' %}
 
-  MathJax.Hub.Config({
-    tex2jax: {
-      inlineMath: [ ['$', '$'], ['\\(', '\\)'] ],
-      processEscapes: true,
-      skipTags: ['script', 'noscript', 'style', 'textarea', 'pre', 'code']
-    },
-    TeX: {
+<script>
+  if (typeof MathJax === 'undefined') {
+    window.MathJax = {
       {%- if theme.math.mathjax.mhchem %}
-        extensions: ['[mhchem]/mhchem.js'],
+        loader: {
+          load: ['[tex]/mhchem']
+        },
       {%- endif %}
-      equationNumbers: {
-        autoNumber: 'AMS'
-      }
-    }
-  });
-
-  MathJax.Hub.Register.StartupHook('TeX Jax Ready', function() {
-    MathJax.InputJax.TeX.prefilterHooks.Add(function(data) {
-      if (data.display) {
-        var next = data.script.nextSibling;
-        while (next && next.nodeName.toLowerCase() === '#text') {
-          next = next.nextSibling;
-        }
-        if (next && next.nodeName.toLowerCase() === 'br') {
-          next.parentNode.removeChild(next);
+      tex: {
+        inlineMath: {'[+]': [['$', '$']]},
+        {%- if theme.math.mathjax.mhchem %}
+          packages: {'[+]': ['mhchem']},
+        {%- endif %}
+        tags: 'ams'
+      },
+      options: {
+        renderActions: {
+          findScript: [10, function (doc) {
+            for (const node of document.querySelectorAll('script[type^="math/tex"]')) {
+              const display = !!node.type.match(/; *mode=display/);
+              const math = new doc.options.MathItem(node.textContent, doc.inputJax[0], display);
+              const text = document.createTextNode('');
+              node.parentNode.replaceChild(text, node);
+              math.start = {node: text, delim: '', n: 0};
+              math.end = {node: text, delim: '', n: 0};
+              doc.math.push(math);
+            }
+          }, '']
         }
       }
-    });
-  });
-
-  MathJax.Hub.Queue(function() {
-    var all = MathJax.Hub.getAllJax(), i;
-    for (i = 0; i < all.length; i += 1) {
-      element = document.getElementById(all[i].inputID + '-Frame').parentNode;
-      if (element.nodeName.toLowerCase() == 'li') {
-        element = element.parentNode;
-      }
-      element.classList.add('has-jax');
-    }
-  });
-</script>
-{%- set mathjax_uri = theme.vendors.mathjax or '//cdn.jsdelivr.net/npm/mathjax@2/MathJax.js?config=TeX-AMS-MML_HTMLorMML' %}
-<script>
-  NexT.utils.getScript('{{ mathjax_uri }}', () => {
-    MathJax.Hub.Typeset();
-  }, window.MathJax);
+    };
+    (function () {
+      var script = document.createElement('script');
+      script.src = '{{ mathjax_uri }}';
+      script.defer = true;
+      document.head.appendChild(script);
+    })();
+  } else {
+    MathJax.texReset();
```

Comment:
http://docs.mathjax.org/en/v3.0-latest/web/typeset.html#resetting-automatic-equation-numbering

Is `MathJax.startup.document.state(0);` required here?

Reply:
done